warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IngestTypeVisitor.java

```diff
@@ -394,32 +388,17 @@ public Set<String> getIngestTypesForFunction(ASTFunctionNode node) {
         FunctionJexlNodeVisitor visitor = FunctionJexlNodeVisitor.eval(node);
-        switch (visitor.namespace()) {
-            case CONTENT_FUNCTION_NAMESPACE:
-                // all content function fields are added
-                ContentFunctionsDescriptor.ContentJexlArgumentDescriptor contentDescriptor = new ContentFunctionsDescriptor().getArgumentDescriptor(node);
-                return contentDescriptor.fieldsAndTerms(Collections.emptySet(), Collections.emptySet(), Collections.emptySet(), null)[0];
-            case EVAL_PHASE_FUNCTION_NAMESPACE:
-                // might be able to exclude certain evaluation phase functions from this step
-                EvaluationPhaseFilterFunctionsDescriptor.EvaluationPhaseFilterJexlArgumentDescriptor evaluationDescriptor = (EvaluationPhaseFilterFunctionsDescriptor.EvaluationPhaseFilterJexlArgumentDescriptor) new EvaluationPhaseFilterFunctionsDescriptor()
-                                .getArgumentDescriptor(node);
-                return evaluationDescriptor.fields(null, Collections.emptySet());
-            case GeoWaveFunctions.GEOWAVE_FUNCTION_NAMESPACE:
-                GeoWaveFunctionsDescriptor.GeoWaveJexlArgumentDescriptor descriptor = (GeoWaveFunctionsDescriptor.GeoWaveJexlArgumentDescriptor) new GeoWaveFunctionsDescriptor()
-                                .getArgumentDescriptor(node);
-                return descriptor.fields(null, Collections.emptySet());
-            case GROUPING_REQUIRED_FUNCTION_NAMESPACE:
-                GroupingRequiredFilterFunctionsDescriptor.GroupingRequiredFilterJexlArgumentDescriptor groupingDescriptor = (GroupingRequiredFilterFunctionsDescriptor.GroupingRequiredFilterJexlArgumentDescriptor) new GroupingRequiredFilterFunctionsDescriptor()
-                                .getArgumentDescriptor(node);
-                return groupingDescriptor.fields(null, Collections.emptySet());
-            case QueryFunctions.QUERY_FUNCTION_NAMESPACE:
-                QueryFunctionsDescriptor.QueryJexlArgumentDescriptor queryDescriptor = (QueryFunctionsDescriptor.QueryJexlArgumentDescriptor) new QueryFunctionsDescriptor()
-                                .getArgumentDescriptor(node);
-                return queryDescriptor.fields(null, Collections.emptySet());
-            default:
-                // do nothing
-                log.warn("Unhandled function namespace: " + visitor.namespace());
+        if (visitor.namespace().equals(CONTENT_FUNCTION_NAMESPACE)) {
+            // all content function fields are added
+            ContentFunctionsDescriptor.ContentJexlArgumentDescriptor contentDescriptor = new ContentFunctionsDescriptor().getArgumentDescriptor(node);
+            return contentDescriptor.fieldsAndTerms(Collections.emptySet(), Collections.emptySet(), Collections.emptySet(), null)[0];
+        } else {
+            JexlArgumentDescriptor descriptor = JexlFunctionArgumentDescriptorFactory.F.getArgumentDescriptor(node);
```

Comment:
Huh?   What happened to all of the other functions descriptors?  Why would you get rid of those?

Reply:
That method call is smart enough to select the correct functions descriptor based on the function namespace and name. 

If the query planner were ever configured with an function descriptor that extended any of the previous descriptors, then this visitor would fail to function correctly.

Reply:
Oh, I see, this is using the argument descriptor factory.... totally missed that. We are good.